File: amstrong.py



n = int(input("N: "))

def check_amstrong(x):
    y = 0
    for i in str(x):
        y += int(i)**n
    if ( y == x ):
        return True
    return False

# space = [i for i in range( 10**(n-1), int( "9"*n ))]
# for i in filter(check_amstrong, space):
#     print(i)

# OR

for i in range( 10**(n-1), int( 10**n )):
    if (check_amstrong( i ) == True):
        print(i)
    else:
        continue

File: amstrong.js
// You can change n to whatever or standardd input
const n = 3;


function check_amstrong(x) {
    // placeholder 
    let y = 0
    // converts to string so that we can iterate over the digits
    xx = x.toString()
    for (let i = 0; i < xx.length; i++ ) {
        y += parseInt(xx[i]) ** n;
    }
    // if the placeholder equals the digit then its an amstrong
    if ( x == y ) {
        return true;
    }
    return false;
}

for ( let i = 10**(n-1); i <= 10**n; i++ ) {
    if (check_amstrong( i ) == true) {
        console.log( i )
    }
}
